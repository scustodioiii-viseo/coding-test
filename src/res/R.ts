import strings from './strings';
import raw from './raw';
import colors from './colors';

const R = {
  strings,
  raw,
  colors,
};

export default R;
