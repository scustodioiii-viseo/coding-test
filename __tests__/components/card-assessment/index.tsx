import * as React from 'react';
import renderer from 'react-test-renderer';
import CardAssessment from 'components/card-assessment';
import R from 'res/R';

const description = '<CardAssessment />';

test(`should render ${description} correctly if has points value`, (): void => {
  const component = renderer
    .create(
      <CardAssessment
        assessment={{
          title: 'Advance Health Screening',
          points: 99999,
          icon: R.raw.assessment,
        }}
      />,
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});

test(`should render ${description} correctly if has no points value`, (): void => {
  const component = renderer
    .create(
      <CardAssessment
        assessment={{
          title: 'Advance Health Screening',
          points: 0,
          icon: R.raw.assessment,
        }}
      />,
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
