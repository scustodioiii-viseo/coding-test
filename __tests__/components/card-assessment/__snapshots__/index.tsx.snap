// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render <CardAssessment /> correctly if has no points value 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 150,
      "margin": 14,
      "padding": 20,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/res/raw/assessment.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 100,
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "marginHorizontal": 12,
        "maxWidth": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Advance Health Screening
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/res/images/icon_arrow_right.png",
      }
    }
    style={
      Object {
        "alignSelf": "flex-end",
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;

exports[`should render <CardAssessment /> correctly if has points value 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 12,
      "flexDirection": "row",
      "height": 150,
      "margin": 14,
      "padding": 20,
    }
  }
>
  <Image
    resizeMode="cover"
    source={
      Object {
        "testUri": "../../../src/res/raw/assessment.png",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "height": 100,
        "width": 75,
      }
    }
  />
  <View
    style={
      Object {
        "marginHorizontal": 12,
        "maxWidth": 150,
      }
    }
  >
    <Text
      style={
        Object {
          "flex": 1,
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Advance Health Screening
    </Text>
    <Text
      style={
        Object {
          "fontSize": 16,
        }
      }
    >
      Earn up to 99999 points
    </Text>
  </View>
  <Image
    source={
      Object {
        "testUri": "../../../src/res/images/icon_arrow_right.png",
      }
    }
    style={
      Object {
        "alignSelf": "flex-end",
        "height": 24,
        "width": 24,
      }
    }
  />
</View>
`;
